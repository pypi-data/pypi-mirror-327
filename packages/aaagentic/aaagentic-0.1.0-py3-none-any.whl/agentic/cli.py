import typer
import os
import requests
from bs4 import BeautifulSoup
from rich.markdown import Markdown
from rich.console import Console
from typing import Optional

from .llm import llm_generate, LLMUsage, CLAUDE_DEFAULT_MODEL, GPT_DEFAULT_MODEL
from .file_cache import file_cache
from .colors import Colors

from agentic.agentic_secrets import agentic_secrets as secrets
from agentic.settings import settings


# Create a state class to hold global options
class State:
    def __init__(self):
        self.no_cache: bool = False


state = State()

# Create the app with a callback
app = typer.Typer()


def quiet_log(*args):
    print(Colors.DARK_GRAY + " ".join(map(str, args)) + Colors.ENDC)


# Add the callback to process global options
@app.callback()
def main(
    no_cache: bool = typer.Option(
        False, "--nocache", help="Disable caching for all commands"
    )
):
    """
    Agentic CLI with various commands for managing secrets and running services.
    """
    state.no_cache = no_cache


@app.command()
def list():
    """List all settings."""
    typer.echo("\n".join(sorted(settings.list_settings())))


@app.command()
def list_secrets():
    """List all secrets."""
    typer.echo("\n".join(sorted(secrets.list_secrets())))


@app.command()
def set(name: str, value: str):
    """Set a setting value."""
    typer.echo(settings.set(name, value))


@app.command()
def set_secret(name: str, value: str):
    """Set a secret."""
    typer.echo(secrets.set_secret(name, value))


@app.command()
def get(name: str):
    """Get a setting."""
    typer.echo(settings.get(name))


@app.command()
def get_secret(name: str):
    """Get a secret."""
    typer.echo(secrets.get_secret(name))


@app.command()
def delete(name: str):
    """Delete a setting."""
    typer.echo(settings.delete_setting(name))


@app.command()
def delete_secret(name: str):
    """Delete a secret."""
    typer.echo(secrets.delete_secret(name))


@app.command()
def ollama():
    """List the latest popular models from Ollama. Use "ollama pull <model> to download."""
    # Download the web page from ollama.com/library
    url = "https://ollama.com/library"
    response = requests.get(url)
    html_content = response.content

    def extract_models(html_page: bytes) -> tuple[str, LLMUsage]:
        # Parse the HTML content using BeautifulSoup
        soup = BeautifulSoup(html_content, "html.parser")

        usage = LLMUsage()
        response = llm_generate(
            """
Get the names and descriptions from the top 20 models in this list:
{{models}}
            """,
            models=(soup.get_text() or ""),
            usage=usage,
        )
        return response, usage

    typer.echo("Current popular Ollama models:")

    usage: LLMUsage | None = None
    if state.no_cache:
        listing, usage = extract_models(html_content)
    else:
        listing = file_cache.get(
            url, ttl_seconds=60 * 12, fetch_fn=lambda: extract_models(html_content)
        )

    console = Console()
    md = Markdown(listing or "")
    console.print(md)
    if usage:
        quiet_log(usage)


@app.command()
def ui():
    """Runs the agentic UI"""
    os.execvp("streamlit", ["streamlit", "run", "src/agentic/ui/chat.py"])


@app.command()
def claude(prompt: str):
    """Runs a completion with Anthropic's Claude sonnet model"""
    usage = LLMUsage()
    typer.echo(llm_generate(prompt, model=CLAUDE_DEFAULT_MODEL, usage=usage))
    quiet_log(usage)


@app.command()
def gpt(
    prompt: str,
    model: str = typer.Option(
        GPT_DEFAULT_MODEL, "--model", help="The model to use for completion"
    ),
):
    """Runs a completion with OpenAI's GPT-4o-mini model"""
    usage = LLMUsage()
    typer.echo(llm_generate(prompt, model=model, usage=usage))
    quiet_log(usage)


@app.command()
def models():
    typer.echo(
        "Visit https://docs.litellm.ai/docs/providers for the full list of models"
    )
    typer.echo(
        """
Popular models:
    openai/o1-mini
    openai/o1-preview
    openai/gpt-4o
    openai/gpt-4o-mini
    anthropic/claude-3-5-sonnet-20240620
    anthropic/claude-3-5-haiku-20241022
     """
    )


@app.command()
def repl(filename: str = typer.Argument(default="", show_default=False)):
    """Runs the agentic REPL"""
    cmd = ["python", "src/agentic/repl.py"]
    if filename:
        cmd.append(filename)
    os.execvp("python", cmd)


if __name__ == "__main__":
    app()
