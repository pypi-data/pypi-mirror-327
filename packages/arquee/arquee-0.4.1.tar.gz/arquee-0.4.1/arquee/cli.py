import subprocess
import sys
from os.path import exists, normpath, join
from os import sep


def start_server():
    print(sys.executable)
    cmd = [
        sys.executable,
        "-m",
        "waitress",
        "serve",
        "arquee.server.web_server:app",
    ]

    with open("server.log", "wb"):
        subprocess.Popen(cmd)

    print("Servidor iniciado en segundo plano en el puerto 8000.")


def create_files():
    origin_example = 'ORIGIN_DB={"host": "host", "port":1433 , "user": "user", "pwd": "password", "db_name": "some_db"}\n'
    destiny_example = 'DESTINY_DB={"host": "host", "port":1433 , "user": "user", "pwd": "password", "db_name": "some_db"}\n'

    if not exists(".env"):
        with open(".env", "a") as file:
            file.write(origin_example)
            file.write(destiny_example)
            print("File .env created successfully")

    if not exists("tables.txt"):
        with open("tables.txt", "a") as file:
            file.write(
                "Here you can specify the tables for create the datalake, delete this line :)"
            )
            print("File tables.txt created successfully")


if __name__ == "__main__":
    create_files()
