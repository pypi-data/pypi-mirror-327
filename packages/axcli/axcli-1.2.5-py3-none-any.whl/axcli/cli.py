import click
from axcli import *

@click.group()
def cli():
    pass

cli.add_command(read)
cli.add_command(load)
cli.add_command(catalog)
cli.add_command(run)
cli.add_command(show)
cli.add_command(live)
