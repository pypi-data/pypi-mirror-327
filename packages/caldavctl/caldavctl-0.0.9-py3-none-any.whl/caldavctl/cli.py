# SPDX-FileCopyrightText: 2024-present Helder Guerreiro <helder@tretas.org>
#
# SPDX-License-Identifier: GPL-3.0-or-later

'''
CalDAV tool
'''

import click

import caldavctl.calendar_dav as calendar
import caldavctl.event as event
import caldavctl.todo as todo
import caldavctl.journal as journal
import caldavctl.ics as ics
import caldavctl.backup as backup

from caldavctl import get_version
from caldavctl.config import Config


def print_version(context, param, value):
    # See https://click.palletsprojects.com/en/stable/options/#callbacks-and-eager-options
    if not value or context.resilient_parsing:
        return
    click.echo(f'caldavctl v{get_version()}')
    context.exit(0)


# General options

@click.group()
@click.option('-c', '--config', 'config_file', envvar='CALDAV_CONFIG', help='Configuration file')
@click.option('--name', envvar='CALDAV_NAME', help='Server nickname')
@click.option('--username', envvar='CALDAV_USERNAME', help='Username on the CalDAV server')
@click.option('--passwd', envvar='CALDAV_PASSWD', help='Password on the CalDAV server')
@click.option('--url', envvar='CALDAV_URL', help='Calendar CalDAV url')
@click.option('--timezone', envvar='CALDAV_TIMEZONE', help='Your time zone')
@click.option('--server', help='Default server (use nickname)')
@click.option('--calendar', help='Default calendar id')
@click.option('--version', is_flag=True, is_eager=True, expose_value=False, callback=print_version,
              help='caldavctl version')
@click.pass_context
def cli_group(context,
              config_file,
              name, username, passwd, url, timezone,
              server, calendar):
    '''
    caldavctl - command line CalDAV client
    '''

    config = Config(
        config_file,
        name,
        username,
        passwd,
        url,
        timezone,
        server,
        calendar
    )

    context.obj = {
        'config': config,
        'option': []
    }


# CALENDAR

@cli_group.group('calendar')
@click.pass_context
def calendar_commands(context):
    '''Commands that deal with the calendars on the server'''
    pass


calendar_commands.add_command(calendar.list_calendars)
calendar_commands.add_command(calendar.create_calendar)
calendar_commands.add_command(calendar.delete_calendar)


# EVENTS

@cli_group.group('event')
@click.pass_context
def event_commands(context):
    '''Event management'''
    pass


event_commands.add_command(event.list_events)
event_commands.add_command(event.create_event)
event_commands.add_command(event.delete_event)


# TODOS

@cli_group.group('todo')
@click.pass_context
def todo_commands(context):
    '''Todo management'''
    pass


todo_commands.add_command(todo.list_todos)
todo_commands.add_command(todo.create_todo)
todo_commands.add_command(todo.delete_todo)
todo_commands.add_command(todo.toggle_todo_complete)
todo_commands.add_command(todo.percentage_complete)


# JOURNALS

@cli_group.group('journal')
@click.pass_context
def journal_commands(context):
    '''Journal management'''
    pass


journal_commands.add_command(journal.list_journals)


# ICS

@cli_group.group('ics')
@click.pass_context
def ics_commands(context):
    '''iCalendar file operations'''
    pass


ics_commands.add_command(ics.get_todo)
ics_commands.add_command(ics.create_todo)
ics_commands.add_command(ics.get_event)


# Backup

@cli_group.group('br')
@click.pass_context
def backup_and_restore_commands(context):
    '''Backup or restore a calendar'''
    pass


backup_and_restore_commands.add_command(backup.backup_calendar)
backup_and_restore_commands.add_command(backup.restore_calendar)


def main():
    cli_group()
