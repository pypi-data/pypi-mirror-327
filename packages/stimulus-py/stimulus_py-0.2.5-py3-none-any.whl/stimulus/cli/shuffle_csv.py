#!/usr/bin/env python3
"""CLI module for shuffling CSV data files."""

import argparse

from stimulus.data.data_handlers import DatasetProcessor


def get_args() -> argparse.Namespace:
    """Get the arguments when using from the commandline.

    Returns:
        Parsed command line arguments.
    """
    parser = argparse.ArgumentParser(description="Shuffle rows in a CSV data file.")
    parser.add_argument(
        "-c",
        "--csv",
        type=str,
        required=True,
        metavar="FILE",
        help="The file path for the csv containing all data",
    )
    parser.add_argument(
        "-y",
        "--yaml",
        type=str,
        required=True,
        metavar="FILE",
        help="The YAML config file that hold all parameter info",
    )
    parser.add_argument(
        "-o",
        "--output",
        type=str,
        required=True,
        metavar="FILE",
        help="The output file path to write the noised csv",
    )

    return parser.parse_args()


def main(data_csv: str, config_yaml: str, out_path: str) -> None:
    """Shuffle the data and split it according to the default split method.

    Args:
        data_csv: Path to input CSV file.
        config_yaml: Path to config YAML file.
        out_path: Path to output shuffled CSV.

    TODO major changes when this is going to select a given shuffle method and integration with split.
    """
    # create a DatasetProcessor object from the config and the csv
    processor = DatasetProcessor(config_path=config_yaml, csv_path=data_csv)

    # shuffle the data with a default seed. TODO get the seed for the config if and when that is going to be set there.
    processor.shuffle_labels(seed=42)

    # save the modified csv
    processor.save(out_path)


def run() -> None:
    """Run the CSV shuffling script."""
    args = get_args()
    main(args.csv, args.yaml, args.output)


if __name__ == "__main__":
    run()
