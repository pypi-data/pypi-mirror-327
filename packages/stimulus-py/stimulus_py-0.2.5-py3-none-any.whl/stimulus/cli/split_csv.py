#!/usr/bin/env python3
"""CLI module for splitting CSV data files."""

import argparse

import yaml

from stimulus.data.data_handlers import DatasetProcessor, SplitManager
from stimulus.data.loaders import SplitLoader
from stimulus.utils.yaml_data import YamlSubConfigDict


def get_args() -> argparse.Namespace:
    """Get the arguments when using from the commandline."""
    parser = argparse.ArgumentParser(description="Split a CSV data file.")
    parser.add_argument(
        "-c",
        "--csv",
        type=str,
        required=True,
        metavar="FILE",
        help="The file path for the csv containing all data",
    )
    parser.add_argument(
        "-y",
        "--yaml",
        type=str,
        required=True,
        metavar="FILE",
        help="The YAML config file that hold all parameter info",
    )
    parser.add_argument(
        "-o",
        "--output",
        type=str,
        required=True,
        metavar="FILE",
        help="The output file path to write the noised csv",
    )
    parser.add_argument(
        "-f",
        "--force",
        type=bool,
        required=False,
        default=False,
        help="Overwrite the split column if it already exists in the csv",
    )

    return parser.parse_args()


def main(data_csv: str, config_yaml: str, out_path: str, *, force: bool = False) -> None:
    """Connect CSV and YAML configuration and handle sanity checks.

    Args:
        data_csv: Path to input CSV file.
        config_yaml: Path to config YAML file.
        out_path: Path to output split CSV.
        force: Overwrite the split column if it already exists in the CSV.
    """
    # create a DatasetProcessor object from the config and the csv
    processor = DatasetProcessor(config_path=config_yaml, csv_path=data_csv)

    # create a split manager from the config
    split_config = processor.dataset_manager.config.split
    with open(config_yaml) as f:
        yaml_config = YamlSubConfigDict(**yaml.safe_load(f))
    split_loader = SplitLoader(seed=yaml_config.global_params.seed)
    split_loader.initialize_splitter_from_config(split_config)
    split_manager = SplitManager(split_loader)

    # apply the split method to the data
    processor.add_split(split_manager=split_manager, force=force)

    # save the modified csv
    processor.save(out_path)


def run() -> None:
    """Run the CSV splitting script."""
    args = get_args()
    main(args.csv, args.yaml, args.output, force=args.force)


if __name__ == "__main__":
    run()
