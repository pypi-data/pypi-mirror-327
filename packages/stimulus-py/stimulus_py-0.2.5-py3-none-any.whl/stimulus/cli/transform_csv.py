#!/usr/bin/env python3
"""CLI module for transforming CSV data files."""

import argparse

import yaml

from stimulus.data.data_handlers import DatasetProcessor, TransformManager
from stimulus.data.loaders import TransformLoader
from stimulus.utils.yaml_data import YamlSubConfigDict


def get_args() -> argparse.Namespace:
    """Get the arguments when using from the commandline."""
    parser = argparse.ArgumentParser(description="CLI for transforming CSV data files using YAML configuration.")
    parser.add_argument(
        "-c",
        "--csv",
        type=str,
        required=True,
        metavar="FILE",
        help="The file path for the csv containing all data",
    )
    parser.add_argument(
        "-y",
        "--yaml",
        type=str,
        required=True,
        metavar="FILE",
        help="The YAML config file that holds all parameter info",
    )
    parser.add_argument(
        "-o",
        "--output",
        type=str,
        required=True,
        metavar="FILE",
        help="The output file path to write the noised csv",
    )

    return parser.parse_args()


def main(data_csv: str, config_yaml: str, out_path: str) -> None:
    """Connect CSV and YAML configuration and handle sanity checks.

    This launcher will be the connection between the csv and one YAML configuration.
    It should also handle some sanity checks.
    """
    # initialize the csv processing class, it open and reads the csv in automatic
    processor = DatasetProcessor(config_path=config_yaml, csv_path=data_csv)

    # initialize the transform manager
    transform_config = processor.dataset_manager.config.transforms
    with open(config_yaml) as f:
        yaml_config = YamlSubConfigDict(**yaml.safe_load(f))
    transform_loader = TransformLoader(seed=yaml_config.global_params.seed)
    transform_loader.initialize_column_data_transformers_from_config(transform_config)
    transform_manager = TransformManager(transform_loader)

    # apply the transformations to the data
    processor.apply_transformation_group(transform_manager)

    # write the transformed data to a new csv
    processor.save(out_path)


def run() -> None:
    """Run the CSV transformation script."""
    args = get_args()
    main(args.csv, args.yaml, args.output)


if __name__ == "__main__":
    run()
