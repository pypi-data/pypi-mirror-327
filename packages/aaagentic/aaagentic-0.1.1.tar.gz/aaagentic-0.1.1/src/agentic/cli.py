import typer
import os
import requests
from bs4 import BeautifulSoup
from rich.markdown import Markdown
from rich.console import Console
from typing import Optional

from .llm import llm_generate, LLMUsage, CLAUDE_DEFAULT_MODEL, GPT_DEFAULT_MODEL
from .file_cache import file_cache
from .colors import Colors

from agentic.agentic_secrets import agentic_secrets as secrets
from agentic.settings import settings


import shutil
from pathlib import Path
from importlib import resources
from typing import Optional
from rich.status import Status


# Create a state class to hold global options
class State:
    def __init__(self):
        self.no_cache: bool = False


state = State()

# Create the app with a callback
app = typer.Typer()


def quiet_log(*args):
    print(Colors.DARK_GRAY + " ".join(map(str, args)) + Colors.ENDC)


# Add the callback to process global options
@app.callback()
def main(
    no_cache: bool = typer.Option(
        False, "--nocache", help="Disable caching for all commands"
    )
):
    """
    Agentic CLI with various commands for managing secrets and running services.
    """
    state.no_cache = no_cache


@app.command()
def list():
    """List all settings."""
    typer.echo("\n".join(sorted(settings.list_settings())))


@app.command()
def list_secrets():
    """List all secrets."""
    typer.echo("\n".join(sorted(secrets.list_secrets())))


@app.command()
def set(name: str, value: str):
    """Set a setting value."""
    typer.echo(settings.set(name, value))


@app.command()
def set_secret(name: str, value: str):
    """Set a secret."""
    typer.echo(secrets.set_secret(name, value))


@app.command()
def get(name: str):
    """Get a setting."""
    typer.echo(settings.get(name))


@app.command()
def get_secret(name: str):
    """Get a secret."""
    typer.echo(secrets.get_secret(name))


@app.command()
def delete(name: str):
    """Delete a setting."""
    typer.echo(settings.delete_setting(name))


@app.command()
def delete_secret(name: str):
    """Delete a secret."""
    typer.echo(secrets.delete_secret(name))


@app.command()
def ollama():
    """List the latest popular models from Ollama. Use "ollama pull <model> to download."""
    # Download the web page from ollama.com/library
    url = "https://ollama.com/library"
    response = requests.get(url)
    html_content = response.content

    def extract_models(html_page: bytes) -> tuple[str, LLMUsage]:
        # Parse the HTML content using BeautifulSoup
        soup = BeautifulSoup(html_content, "html.parser")

        usage = LLMUsage()
        response = llm_generate(
            """
Get the names and descriptions from the top 20 models in this list:
{{models}}
            """,
            models=(soup.get_text() or ""),
            usage=usage,
        )
        return response, usage

    typer.echo("Current popular Ollama models:")

    usage: LLMUsage | None = None
    if state.no_cache:
        listing, usage = extract_models(html_content)
    else:
        listing = file_cache.get(
            url, ttl_seconds=60 * 12, fetch_fn=lambda: extract_models(html_content)
        )

    console = Console()
    md = Markdown(listing or "")
    console.print(md)
    if usage:
        quiet_log(usage)


@app.command()
def ui():
    """Runs the agentic UI"""
    os.execvp("streamlit", ["streamlit", "run", "src/agentic/ui/chat.py"])


@app.command()
def claude(prompt: str):
    """Runs a completion with Anthropic's Claude sonnet model"""
    usage = LLMUsage()
    typer.echo(llm_generate(prompt, model=CLAUDE_DEFAULT_MODEL, usage=usage))
    quiet_log(usage)


@app.command()
def gpt(
    prompt: str,
    model: str = typer.Option(
        GPT_DEFAULT_MODEL, "--model", help="The model to use for completion"
    ),
):
    """Runs a completion with OpenAI's GPT-4o-mini model"""
    usage = LLMUsage()
    typer.echo(llm_generate(prompt, model=model, usage=usage))
    quiet_log(usage)


@app.command()
def models():
    typer.echo(
        "Visit https://docs.litellm.ai/docs/providers for the full list of models"
    )
    typer.echo(
        """
Popular models:
    openai/o1-mini
    openai/o1-preview
    openai/gpt-4o
    openai/gpt-4o-mini
    anthropic/claude-3-5-sonnet-20240620
    anthropic/claude-3-5-haiku-20241022
     """
    )


@app.command()
def repl(filename: str = typer.Argument(default="", show_default=False)):
    """Runs the agentic REPL"""
    cmd = ["python", "src/agentic/repl.py"]
    if filename:
        cmd.append(filename)
    os.execvp("python", cmd)


def copy_examples(src_path: Path, dest_path: Path, console: Console) -> None:
    """Copy example files from source to destination, maintaining directory structure."""
    try:
        if not dest_path.exists():
            dest_path.mkdir(parents=True)
        
        # Copy all files and directories
        for item in src_path.iterdir():
            target = dest_path / item.name
            
            if item.is_file():
                shutil.copy2(item, target)
                console.print(f"‚úì Copied: {item.name}", style="green")
            elif item.is_dir():
                shutil.copytree(item, target, dirs_exist_ok=True)
                console.print(f"‚úì Copied directory: {item.name}", style="green")
                
    except Exception as e:
        console.print(f"Error copying examples: {str(e)}", style="red")
        raise typer.Exit(1)

@app.command()
def init(
    path: str = typer.Argument(
        ".", help="Directory to initial your project (defaults to current directory)"
    ),
    force: bool = typer.Option(
        False,
        "--force",
        "-f",
        help="Overwrite existing files in the destination directory"
    )
):
    """Initialize a new project by copying example files from the package."""
    console = Console()
    dest_path = Path(path + "/examples").resolve()
    os.mkdir("agents")
    os.mkdir("tools")
    os.mkdir("evals")

    # Check if destination exists and is not empty
    if dest_path.exists() and any(dest_path.iterdir()) and not force:
        console.print(
            "‚ö†Ô∏è  Destination directory is not empty. Use --force to overwrite existing files.",
            style="yellow"
        )
        raise typer.Exit(1)
    
    with Status("[bold green]Copying example files...", console=console):
        try:
            # Get the package's examples directory using importlib.resources
            # Replace 'your_package_name' with your actual package name
            with resources.path('agentic_examples', '') as examples_path:
                copy_examples(examples_path, dest_path, console)
            
            console.print("\n‚ú® Examples copied successfully!", style="bold green")
            console.print(f"üìÅ Location: {dest_path}", style="blue")
            
        except ModuleNotFoundError:
            console.print("Error: Could not find examples directory in package.", style="red")
            raise typer.Exit(1)
        except Exception as e:
            console.print(f"Error initializing project: {str(e)}", style="red")
            raise typer.Exit(1)
        

if __name__ == "__main__":
    app()
