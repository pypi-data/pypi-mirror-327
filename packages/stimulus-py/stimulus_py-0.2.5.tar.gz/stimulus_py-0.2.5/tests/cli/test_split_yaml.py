"""Tests for the split_yaml CLI command."""

import hashlib
import os
import tempfile
from typing import Any, Callable

import pytest

from src.stimulus.cli.split_yaml import main


# Fixtures
@pytest.fixture
def correct_yaml_path() -> str:
    """Fixture that returns the path to a correct YAML file."""
    return "tests/test_data/titanic/titanic.yaml"


@pytest.fixture
def wrong_yaml_path() -> str:
    """Fixture that returns the path to a wrong YAML file."""
    return "tests/test_data/yaml_files/wrong_field_type.yaml"


# Test cases
test_cases = [
    ("correct_yaml_path", None),
    ("wrong_yaml_path", ValueError),
]


# Tests
@pytest.mark.parametrize(("yaml_type", "error"), test_cases)
def test_split_yaml(
    request: pytest.FixtureRequest,
    snapshot: Callable[[], Any],
    yaml_type: str,
    error: Exception | None,
) -> None:
    """Tests the CLI command with correct and wrong YAML files."""
    yaml_path = request.getfixturevalue(yaml_type)
    tmpdir = tempfile.gettempdir()
    if error:
        with pytest.raises(error):  # type: ignore[call-overload]
            main(yaml_path, tmpdir)
    else:
        main(yaml_path, tmpdir)  # main() returns None, no need to assert
        files = os.listdir(tmpdir)
        test_out = [f for f in files if f.startswith("test_")]
        hashes = []
        for f in test_out:
            with open(os.path.join(tmpdir, f)) as file:
                hashes.append(hashlib.md5(file.read().encode()).hexdigest())  # noqa: S324
        assert sorted(hashes) == snapshot  # sorted ensures that the order of the hashes does not matter
