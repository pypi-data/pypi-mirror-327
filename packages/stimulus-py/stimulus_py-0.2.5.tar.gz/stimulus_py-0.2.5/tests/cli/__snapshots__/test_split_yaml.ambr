# serializer version: 1
# name: test_split_yaml[correct_yaml_path-None]
  list([
    '0e43b7cdcd8d458cc4e6ff80e06ba7ea',
    '43a7f9fbac5c32f51fa51680c7679a57',
    'edf8dd2d39b74619d17b298e3b010c77',
  ])
# ---
